shared/types/field/dropdownField.ts

```diff
@@ -3,4 +3,5 @@ import { BasicField, MyInfoableFieldBase } from './base'
 export interface DropdownFieldBase extends MyInfoableFieldBase {
   fieldType: BasicField.Dropdown
   fieldOptions: string[]
+  optionsToRecipientsMap?: Record<string, string[]>
```

Comment:
As discussed, we bind the mapping to the dropdown field. The avoids the issue where the step is not yet created when trying to bind the mapping to the step (if the options to recipients mapping is stored in the workflow step instead)

Reply:
I feel that this is an important context that we want to either place it in our docs or in code.

This also helps to avoid the issue where we have stragglers when the stepID is deleted. We can avoid the problems of needing to cascade field deletion when this Dropdown field is deleted on the CreateAndBuilder Page.

Reply:
documented in [d9e4aa6](https://github.com/opengovsg/FormSG/pull/7804/commits/d9e4aa696a334bc1e285ef06e154b3f9b025ac81)